patchwork-fml/src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java

```diff
@@ -0,0 +1,340 @@
+		try {
+			findField(classToAccess, fieldIndex).set(instance, value);
+		} catch (IllegalAccessException e) {
+			LOGGER.error("There was a problem setting field index {} on type {}", fieldIndex, classToAccess.getName(), e);
+			throw new UnableToAccessFieldException(e);
+		}
+	}
+
+	/**
+	 * Sets the value a field with the specified name in the given class.
+	 * Note: For performance, use {@link #findField(Class, String)} if you are setting the value more than once.
+	 *
+	 * <p>Throws an exception if the field is not found or the value of the field cannot be set.</p>
+	 *
+	 * @param classToAccess The class to find the field on.
+	 * @param instance      The instance of the {@code classToAccess}.
+	 * @param value         The new value for the field
+	 * @param fieldName     The name of the field in the {@code classToAccess}.
+	 * @param <T>           The type of the value.
+	 * @param <E>           The type of the {@code classToAccess}.
+	 * @throws UnableToFindFieldException   If there was a problem getting the field.
+	 * @throws UnableToAccessFieldException If there was a problem setting the value of the field.
+	 */
+	public static <T, E> void setPrivateValue(@Nonnull final Class<? super T> classToAccess, @Nonnull final T instance,
+				@Nullable final E value, @Nonnull final String fieldName) {
+		try {
+			findField(classToAccess, fieldName).set(instance, value);
+		} catch (UnableToFindFieldException e) {
+			LOGGER.error("Unable to locate any field {} on type {}", fieldName, classToAccess.getName(), e);
+			throw e;
+		} catch (IllegalAccessException e) {
+			LOGGER.error("Unable to set any field {} on type {}", fieldName, classToAccess.getName(), e);
+			throw new UnableToAccessFieldException(e);
+		}
+	}
+
+	/**
+	 * Finds a method with the specified name and parameters in the given class and makes it accessible.
+	 * Note: For performance, store the returned value and avoid calling this repeatedly.
+	 *
+	 * <p>Throws an exception if the method is not found.</p>
+	 *
+	 * @param clazz          The class to find the method on.
+	 * @param methodName     The intermediary (unmapped) name of the method to find (e.g. "method_342").
+	 * @param parameterTypes The parameter types of the method to find.
+	 * @return The method with the specified name and parameters in the given class.
+	 * @throws NullPointerException        If {@code clazz} is null.
+	 * @throws NullPointerException        If {@code methodName} is null.
+	 * @throws IllegalArgumentException    If {@code methodName} is empty.
+	 * @throws NullPointerException        If {@code parameterTypes} is null.
+	 * @throws UnableToFindMethodException If the method could not be found.
+	 */
+	@Nonnull
+	public static Method findMethod(@Nonnull final Class<?> clazz, @Nonnull final String methodName, @Nonnull final Class<?>... parameterTypes) {
+		Preconditions.checkNotNull(clazz, "Class to find method on cannot be null.");
```

Comment:
Can't this be replaced with `Objects.requireNonNull`?

Reply:
Technically, yes, but IMO it's better for consistency because methods that have no equivalent in Objects like `Preconditions.checkArgument` are used elsewhere in this class.